src/lib/recurringPurposes.js

```diff
@@ -367,7 +368,37 @@ const queryRecurringTimeseriesWithCloseStartOrEnd = async (
   return results || []
 }
 
-// Similar trips = close start/end point
+export const filterTripsBasedOnDistance = (timeseries, baseDistance) => {
+  const maxDistance =
+    baseDistance + baseDistance * TRIPS_DISTANCE_SIMILARITY_RATIO
+  const minDistance =
+    baseDistance - baseDistance * TRIPS_DISTANCE_SIMILARITY_RATIO
+  return timeseries.filter(ts => {
+    return (
+      ts.aggregation?.totalDistance >= minDistance &&
+      ts.aggregation?.totalDistance <= maxDistance
```

Comment:
Since we have detected that we're in a `LoopTrip`, should we really check that we have an aggregation? Should not be always the case? 

At some time, maybe it can be useful to create a new type with the aggregation attribute mandatory, and cast to that type when we know this is true, like that, we can stop doing `?.` everywhere

Reply:
Fair enough, I tend to automate the use of `?.` for safety